packages/mobile/src/services/offline-downloader/workers/collectionDownloadWorker.ts

```diff
@@ -1,13 +1,41 @@
 import type { CancellablePromise } from 'react-native-job-queue'
-import { Worker } from 'react-native-job-queue'
+import queue, { Worker } from 'react-native-job-queue'
 
-import { downloadCollection } from '../offline-downloader'
+import { isEqualCollectionPayload } from 'app/services/offline-downloader/offline-download-queue'
+
+import {
+  downloadCollection,
+  DownloadCollectionError
+} from '../offline-downloader'
 import type { CollectionForDownload } from '../types'
 
 export const COLLECTION_DOWNLOAD_WORKER = 'collection_download_worker'
 export type CollectionDownloadWorkerPayload = CollectionForDownload
 
-const onFailure = ({ payload }: { payload: CollectionForDownload }) => {}
+const onFailure = async (
+  { payload }: { payload: CollectionForDownload },
+  error: Error
+) => {
+  switch (error.message) {
```

Comment:
woah, this is really nifty, so if im understanding, if a worker fails out, which we are doing for collection worker, `onFailure` is called, and we remove the job for IS_DELETED and IS_PRIVATE so it doesn't try those cases again?

Reply:
yee